pkg/vsphere/vsphere.go

```diff
@@ -500,6 +501,132 @@ func (v *VSphereCloudDriver) getClusterComputeResources(ctx context.Context, fin
 	return ccrs, nil
 }
 
+type hostDateInfo struct {
+	hostName   string
+	ntpServers []string
+	types.HostDateTimeInfo
+	Service       *types.HostService
+	Current       *time.Time
+	clientStatus  string
+	serviceStatus string
+}
+
+func (info *hostDateInfo) servers() []string {
+	return info.NtpConfig.Server
+}
+
+func (v *VSphereCloudDriver) ValidateHostNTPSettings(ctx context.Context, driver *VSphereCloudDriver, finder *find.Finder, datacenter string, clusterName string, hosts []string) (isvalid bool, failures []string, err error) {
```

Comment:
Tidy the signature please:

`ValidateHostNTPSettings(ctx context.Context, driver *VSphereCloudDriver, finder *find.Finder, datacenter, clusterName string, hosts []string) (bool, []string, error)`

Removed named return vars as you're not using them properly (exiting via `return` only) and they should be avoided most of the time due to the likelihood of variable shadowing bugs.